communication/src/TlsConnectionManager.cpp

```diff
@@ -58,24 +58,32 @@ void ConnectionManager::stop() {
   stopped_ = true;
   status_->reset();
   acceptor_.close();
+
   for (auto& [_, sock] : connecting_) {
     (void)_;  // unused variable hack
     sock.first.close();
   }
+  connecting_.clear();
```

Comment:
I suggest to add a task under out epic to convert socket (file) handling into smart pointer with custime deleter. Theses close are C style.
if we have custim deleter, you only need to clear.
Same for all variations down there, they are all can ve converted to smart pointers.

Reply:
I have no objection, but it's outside of this PR's scope as I see it